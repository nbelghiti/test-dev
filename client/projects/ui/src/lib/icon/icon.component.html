<i class="c-icon__glyph {{ size && 'c-icon__glyph--' + size }} {{ type && 'c-icon__glyph--' + type }} fas fa-{{ name }}"></i>
