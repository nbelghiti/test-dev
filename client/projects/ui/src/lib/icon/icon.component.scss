// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Icon
// =============================================================================
:host {
	display: inline-flex;
	justify-content: center;
	align-items: center;

	font-size: inherit;
}

// =============================================================================
// :: Elements
// =============================================================================
.c-icon__glyph {
	line-height: inherit;

	cursor: pointer;
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-icon__glyph--large {
	font-size: 2rem;
}

.c-icon__glyph--small {
	font-size: 1.4rem;
}

.c-icon__glyph--tiny {
	font-size: 1.2rem;
}

.c-icon__glyph--success {
	color: $ui-color-success-dark;
}

.c-icon__glyph--error {
	color: $ui-color-error-dark;
}

.c-icon__glyph--warning {
	color: $ui-color-warning-dark;
}

.c-icon__glyph--info {
	color: $ui-color-info-dark;
}
