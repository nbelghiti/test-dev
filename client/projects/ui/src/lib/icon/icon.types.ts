export enum IconName {
	archive = 'archive',
	arrowDown = 'arrow-down',
	arrowLeft = 'arrow-left',
	arrowRight = 'arrow-right',
	arrowRightAlt = 'arrow-alt-circle-right',
	arrows = 'arrows-alt-h',
	arrowUp = 'arrow-up',
	bell = 'bell',
	bellSlash = 'bell-slash',
	book = 'book',
	brackets = 'brackets',
	calendar = 'calendar',
	caretDown = 'caret-down',
	caretLeft = 'caret-left',
	caretRight = 'caret-right',
	caretUp = 'caret-up',
	check = 'check',
	checkCircle = 'check-circle',
	chevronDown = 'chevron-down',
	chevronLeft = 'chevron-left',
	chevronRight = 'chevron-right',
	chevronUp = 'chevron-up',
	clock = 'clock',
	clone = 'clone',
	close = 'times-circle',
	code = 'code',
	codeBranch = 'code-branch',
	cube = 'cube',
	download = 'download',
	edit = 'edit',
	excel = 'file-excel',
	exclamation = 'exclamation',
	exclamationCircle = 'exclamation-circle',
	expand = 'expand',
	externalLink = 'external-link-alt',
	file = 'file-alt',
	home = 'home',
	horizontalEllipsis = 'ellipsis-h',
	industry = 'industry',
	infoCircle = 'info-circle',
	invisible = 'eye-slash',
	marker = 'map-marker-alt',
	minus = 'minus',
	pdf = 'file-pdf',
	pen = 'pen',
	plus = 'plus',
	print = 'print',
	search = 'search',
	settings = 'cog',
	signOut = 'sign-out-alt',
	snowflake = 'snowflake',
	spinner = 'spinner',
	star = 'star',
	sync = 'sync-alt',
	table = 'table',
	times = 'times',
	trash = 'trash',
	upload = 'upload',
	user = 'user',
	users = 'users',
	visible = 'eye',
	plusSquare = 'plus-square',
	minusSquare = 'minus-square',
}

export enum IconType {
	success = 'success',
	error = 'error',
	warning = 'warning',
	info = 'info',
}

export enum IconSize {
	large = 'large',
	small = 'small',
	tiny = 'tiny',
}
