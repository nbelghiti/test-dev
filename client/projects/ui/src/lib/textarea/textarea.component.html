<div
	class="c-textarea"
	[ngClass]="{
		'c-textarea--disabled': disabled,
		'c-textarea--error': error
	}"
>
	<textarea
		[id]="name"
		[name]="name"
		class="c-textarea__field"
		[placeholder]="placeholder"
		[disabled]="disabled"
		[rows]="rows"
		[ngModel]="value"
		(ngModelChange)="writeValue($event)"
		required
	>
	</textarea>

	<label
		class="c-textarea__label"
		[for]="name"
	>
		{{label}}
	</label>

	<div *ngIf="error" class="c-error">
		<ui-icon
			name="exclamation-circle"
			size="small"
		>
		</ui-icon>

		<span class="c-error__text">
			{{error}}
		</span>
	</div>
</div>
