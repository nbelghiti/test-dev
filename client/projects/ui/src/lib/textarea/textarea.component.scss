// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Textarea
// =============================================================================
.c-textarea {
	position: relative;

	display: flex;
	flex-direction: column;

	background: none;
}

// =============================================================================
// :: Elements
// =============================================================================
.c-textarea__label {
	position: absolute;
	top: 0;
	left: 1rem;

	order: -1;

	font-size: $font-size-1x-small;
	line-height: $line-height-base;

	color: $text-color-grey-56;

	transform-origin: left center;
	transform: scale(1.15) translateY(1rem);

	transition: transform $animation-base;

	pointer-events: none;
}

.c-textarea__field {
	display: inline-block;
	padding: 2.4rem 2rem 0.2rem;

	font-size: $font-size-base;
	line-height: $line-height-base;

	background: none;
	border: $border-width-base solid $border-color-grey-76;
	color: $text-color-base;

	transition:
		border $animation-base,
		box-shadow $animation-base;

	resize: none;
	cursor: text;

	&:focus {
		border: $border-width-base solid $text-color-base;
		box-shadow: 0 0 0 0.3rem rgba($box-shadow-color-base, 0.1);
	}

	&::placeholder {
		color: $text-color-grey-56;

		opacity: 0;

		transition: all $animation-base;

		visibility: hidden;
	}

	&:focus,
	&:valid {

		+ .c-textarea__label {
			transform: scale(1) translateY(0);
		}

		&::placeholder {

			opacity: 1;

			visibility: visible;
		}
	}
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-textarea--disabled {
	cursor: not-allowed;

	.c-textarea__field {
		background-color: $background-color-grey-96;
		color: $text-color-grey-46;

		pointer-events: none;
	}

	.c-textarea__label {
		cursor: not-allowed;
	}
}

.c-textarea--error {

	.c-textarea__field {
		border: $border-width-base solid $ui-color-error-dark;
	}
}
