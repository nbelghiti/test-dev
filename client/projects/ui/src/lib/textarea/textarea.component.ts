import { Component, Input, forwardRef } from '@angular/core';
import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
import { isEmpty } from 'ramda';

@Component({
	selector: 'ui-textarea',
	templateUrl: './textarea.component.html',
	styleUrls: ['./textarea.component.scss'],
	providers: [
		{
			provide: NG_VALUE_ACCESSOR,
			useExisting: forwardRef(() => TextareaComponent),
			multi: true,
		},
	],
})

export class TextareaComponent implements ControlValueAccessor {
	@Input() public name: string;
	@Input() public placeholder: string = '';
	@Input() public label: string;
	@Input() public rows: number = 4;
	@Input() public disabled?: boolean = false;
	@Input() public error?: string = '';

	public value: string;

	public onChange: (_) => any = () => { };

	public onTouched = () => { };

	public writeValue(value: string): void {
		this.value = value;

		if (isEmpty(value)) {
			this.value = null;
		}

		this.onChange(this.value);
	}

	public registerOnChange(onChange: (_) => any): void {
		this.onChange = onChange;
	}

	public registerOnTouched(fn: () => void): void {
		this.onTouched = fn;
	}

	public setDisabledState(isDisabled: boolean): void {
		this.disabled = isDisabled;
	}
}
