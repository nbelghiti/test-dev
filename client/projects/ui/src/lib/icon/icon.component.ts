import { Component, HostBinding, Input } from '@angular/core';

import { IconName, IconSize, IconType } from './icon.types';

@Component({
	selector: 'ui-icon',
	templateUrl: './icon.component.html',
	styleUrls: ['./icon.component.scss']
})

export class IconComponent {
	@Input() public name: IconName;
	@Input() public type: IconType = null;
	@Input() public size: IconSize = null;

	@HostBinding('class.c-icon') public className = true;
}
