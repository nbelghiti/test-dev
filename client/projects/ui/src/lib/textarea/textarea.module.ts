import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';

import { IconModule } from '../icon/icon.module';

import { TextareaComponent } from './textarea.component';

@NgModule({
	imports: [
		CommonModule,
		IconModule,
		FormsModule
	],
	declarations: [
		TextareaComponent,
	],
	exports: [
		TextareaComponent,
	],
})
export class TextareaModule {}
