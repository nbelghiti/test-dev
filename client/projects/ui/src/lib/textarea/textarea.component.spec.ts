import { ComponentFixture, TestBed, async } from '@angular/core/testing';
import { FormsModule } from '@angular/forms';

import { IconComponent } from '../icon/icon.component';

import { TextareaComponent } from './textarea.component';

describe('TextareaComponent', () => {
	let component: TextareaComponent;
	let fixture: ComponentFixture<TextareaComponent>;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			imports: [
				FormsModule,
			],
			declarations: [
				IconComponent,
				TextareaComponent
			]
		})
		.compileComponents();
	}));

	beforeEach(() => {
		fixture = TestBed.createComponent(TextareaComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('Should render correctly', () => {
		expect(fixture).toMatchSnapshot();
	});
});
