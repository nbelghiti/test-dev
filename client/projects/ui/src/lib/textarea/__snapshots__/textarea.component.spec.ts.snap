// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextareaComponent Should render correctly 1`] = `
<ui-textarea
  disabled="false"
  error=""
  onChange={[Function Function]}
  onTouched={[Function Function]}
  placeholder=""
  rows={[Function Number]}
>
  <div
    class="c-textarea"
    ng-reflect-klass="c-textarea"
    ng-reflect-ng-class="[object Object]"
  >
    <textarea
      class="c-textarea__field ng-untouched ng-pristine ng-invalid"
      id="undefined"
      ng-reflect-is-disabled="false"
      ng-reflect-required=""
      placeholder=""
      required=""
      rows="4"
    >
      	
    </textarea>
    <label
      class="c-textarea__label"
      for="undefined"
    >
        
    </label>
    
  </div>
</ui-textarea>
`;
